bookkeeper-server/src/main/java/org/apache/bookkeeper/tls/TLSContextFactory.java

```diff
@@ -262,7 +269,36 @@ private void createClientContext(AbstractConfiguration conf)
         sslContext = sslContextBuilder.build();
     }
 
-    private void createServerContext(AbstractConfiguration conf) throws SecurityException, KeyStoreException,
+    private void createServerContext()
+            throws SecurityException, KeyStoreException, NoSuchAlgorithmException, CertificateException, IOException,
+            UnrecoverableKeyException, InvalidKeySpecException, IllegalArgumentException {
+        isServerCtx = true;
+        ServerConfiguration serverConf = (ServerConfiguration) config;
+        tTLSCertificatePath = new FileModifiedTimeUpdater(serverConf.getTLSCertificatePath());
+        tLSKeyStoreFilePath = new FileModifiedTimeUpdater(serverConf.getTLSKeyStore());
+        tLSKeyStorePasswordFilePath = new FileModifiedTimeUpdater(serverConf.getTLSKeyStorePasswordPath());
+        tLSTrustStoreFilePath = new FileModifiedTimeUpdater(serverConf.getTLSTrustStore());
+        tLSTrustStorePasswordFilePath = new FileModifiedTimeUpdater(serverConf.getTLSTrustStorePasswordPath());
+        updateServerContext();
+    }
+
+    private SslContext getSSLContext() {
```

Comment:
synchronized? Otherwise we will have a race condition for certRefreshTime

Reply:
Sorry I meant certLastRefreshTime.
It can happen that 2 threads enter this method and start creating a new context.
If you don't want to make this method synchronized you have to double check for the condition on certLastRefreshTime inside the critical section